src/lib/jobs/deep-research.ts

```diff
@@ -0,0 +1,236 @@
+    return [];
+  }
+}
+
+/**
+ * Execute multiple search queries in parallel with higher limits
+ * and return combined results
+ */
+async function executeDeepSearchQueries(
+  db: DrizzleDB,
+  userId: string,
+  queries: string[]
+): Promise<string> {
+  // Deep search uses higher limits than real-time search
+  const deepSearchLimit = 20;
+  
+  // Run all search queries in parallel
+  const searchResults = await Promise.all(
+    queries.map(async (query) => {
+      const embedding = await generateQueryEmbedding(query);
+      if (!embedding) return null;
+      
+      return executeSearchWithEmbedding(db, userId, query, embedding, deepSearchLimit);
+    })
+  );
+  
+  // Combine all search results
+  const combinedResult = searchResults
+    .filter(Boolean)
+    .join("\n");
+    
+  return combinedResult || "";
+}
+
+/**
+ * Generate embedding for search query
+ */
+async function generateQueryEmbedding(query: string): Promise<number[] | null> {
+  try {
+    const embeddingsResponse = await generateEmbeddings({
+      model: "jina-embeddings-v3",
+      task: "retrieval.query",
+      input: [query],
+      truncate: true,
+    });
+    
+    return embeddingsResponse.data[0]?.embedding || null;
+  } catch (error) {
+    console.error("Failed to generate embedding for query:", error);
+    return null;
+  }
+}
+
+/**
+ * Execute a single search with the provided embedding
+ */
+async function executeSearchWithEmbedding(
+  db: DrizzleDB,
+  userId: string,
+  query: string,
+  embedding: number[],
+  limit: number
+): Promise<string | null> {
+  try {
+    // Find similar nodes and edges
+    const [similarNodes, similarEdges] = await Promise.all([
+      findSimilarNodes({
+        userId,
+        embedding,
+        limit,
+        minimumSimilarity: 0.35, // Lower threshold for deep search
+      }),
+      findSimilarEdges({
+        userId,
+        embedding,
+        limit,
+        minimumSimilarity: 0.35, // Lower threshold for deep search
+      }),
+    ]);
+
+    // Get one-hop connections
+    const nodeIds = new Set([
+      ...similarNodes.map((node) => node.id),
+      ...similarEdges.flatMap((edge) => [edge.sourceNodeId, edge.targetNodeId]),
+    ]);
+
+    const connections = await findOneHopNodes(db, userId, Array.from(nodeIds));
+
+    // Rerank results
+    const rerankedResults = await rerankMultiple(
+      query,
+      {
+        similarNodes: {
+          items: similarNodes,
+          toDocument: (n) => `${n.label}: ${n.description}`,
+        },
+        similarEdges: {
+          items: similarEdges,
+          toDocument: (e) =>
+            `${e.sourceLabel ?? ""} -> ${e.targetLabel ?? ""}: ${e.edgeType}` +
+            (e.description ? `: ${e.description}` : ""),
+        },
+        connections: {
+          items: connections,
+          toDocument: (c) => `${c.label}: ${c.description}`,
+        },
+      },
+      limit,
+    );
+
+    return formatSearchResultsAsXml(rerankedResults);
```

Comment:
We should probably not format them to string here, so that we can do a *single* format as XML call that runs on both the immediate query results and the deep search results at the same time. That way we don't have two different XML blocks.